pkg/kubediscovery/handlers.go

```diff
@@ -63,75 +41,137 @@ type caLoader interface {
 // fsCALoader is a caLoader for loading the PEM encoded CA from
 // /tmp/secret/ca.pem.
 type fsCALoader struct {
+	certData string
 }
 
 func (cl *fsCALoader) LoadPEM() (string, error) {
-	file, err := os.Open(CAPath)
-	if err != nil {
-		return "", err
+	if cl.certData != "" {
+		data, err := ioutil.ReadFile(CAPath)
+		if err != nil {
+			return "", err
+		}
+
+		cl.certData = base64.StdEncoding.EncodeToString(data)
```

Comment:
In my testing I assume we start with a PEM encoded cert, we must b64 encode it and place it in the secret for k8s. However inside the pod it will appear on disk in it's original PEM format.

Is it ok if I drop this b64 re-encode here, I think it's unnecessary as when we go do our JWS signing, we convert the whole data blog to b64 anyhow, so I think we can safely skip this one here and just send back the PEM. Sound ok?